### PR TITLE
Fixes NE DS doorgun deployment bug

### LINKED ISSUE #5236: MG deploys in inaccessible way on Dropship NE weapons hard point
### Testmerges

#4842: '/tg/ Status Effects Part 2 - datum, KD, KO, Stuns' by fira at commit 595086537a #5007: 'Project ARES TM Holder (v4)' by realforest2001 at commit 3552058523 #4477: 'Keycard authenticator can call ERT winout admin intervention' by Diegoflores31 at commit 8f625da736 #5204: 'Requisitions elevator lighting fix' by NateDross at commit 12b7a0d07b #5202: 'Adds a way for SLs/SOs/XOs to recommend medals that the CO can then approve/deny.' by TheGamerdk at commit 1ebcc431cb #5219: 'Ghost listening device preference' by realforest2001 at commit afd02c87de #5235: 'Adds closeother to USS Almayer doors.' by realforest2001 at commit 2fd94702ed

### Round ID

19994

### Description of the bug

image

### What's the difference with what should have happened?

SHould deploy in the other direction.

### How do we reproduce this bug?

1.purchase MG hardpoint, put on NE hardpoint and deploy, NW works.
2.
3.
...

### Issue Bingo

- [X] Issue could be reproduced at least once
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] Couldn't find an existing issue about this (https://github.com/cmss13-devs/cmss13/issues)

### PR DESCRIPTION
**Pull request by cuberound**
# About the pull request
fixes #5236
fixes the issue that makes DS doorgun deploy itself into DS when placed on NE gunslot

# Explain why it's good for the game

doorgun in wall bad

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
fix: bad DS doorgun placement
/:cl:

### CODE REVIEW

File: code/modules/cm_marines/dropship_equipment.dm [hunk 1]
```diff
@@ -339,9 +339,10 @@
 			if(ship_base.base_category == DROPSHIP_WEAPON)
 				switch(dir)
 					if(NORTH)
-						if( istype(get_step(src, WEST), /turf/open) )
+						var step_contents = get_step(src, EAST).contents
```
Comment (harryob) [inline code/modules/cm_marines/dropship_equipment.dm:342]:
```suggestion
						var/step_contents = get_step(src, EAST).contents
```

File: code/modules/cm_marines/dropship_equipment.dm [hunk 2]
```diff
@@ -368,12 +369,11 @@
 		if(ship_base.base_category == DROPSHIP_WEAPON)
 			switch(dir)
 				if(NORTH)
-					if( istype(get_step(src, WEST), /turf/open) )
+					var step_contents = get_step(src, EAST).contents
```
Comment (harryob) [inline code/modules/cm_marines/dropship_equipment.dm:372]:
```suggestion
					var/step_contents = get_step(src, EAST).contents
```

**Review by harryob — CHANGES_REQUESTED** [review]
otherwise fine

**Review by Drulikar — APPROVED** [review]
Works:
image

**Pull request merged**